packages/server/src/sdk/app/rowActions.ts

```diff
@@ -97,55 +100,97 @@ export async function docExists(tableId: string) {
   return result
 }
 
-export async function update(
+async function updateDoc(
   tableId: string,
   rowActionId: string,
-  rowAction: { name: string }
+  transformer: (
+    tableRowActions: TableRowActions
+  ) => TableRowActions | Promise<TableRowActions>
 ) {
-  const action = { name: rowAction.name.trim() }
   const actionsDoc = await get(tableId)
-
-  if (!actionsDoc.actions[rowActionId]) {
+  const rowAction = actionsDoc?.actions[rowActionId]
+  if (!rowAction) {
     throw new HTTPError(
       `Row action '${rowActionId}' not found in '${tableId}'`,
       400
     )
   }
 
-  ensureUniqueAndThrow(actionsDoc, action.name, rowActionId)
-
-  actionsDoc.actions[rowActionId] = {
-    automationId: actionsDoc.actions[rowActionId].automationId,
-    ...action,
-  }
+  const updated = await transformer(actionsDoc)
 
   const db = context.getAppDB()
-  await db.put(actionsDoc)
+  await db.put(updated)
 
   return {
     id: rowActionId,
-    ...actionsDoc.actions[rowActionId],
+    ...updated.actions[rowActionId],
   }
 }
 
-export async function remove(tableId: string, rowActionId: string) {
-  const actionsDoc = await get(tableId)
+export async function update(
+  tableId: string,
+  rowActionId: string,
+  rowActionData: { name: string }
+) {
+  const newName = rowActionData.name.trim()
 
-  const rowAction = actionsDoc.actions[rowActionId]
-  if (!rowAction) {
-    throw new HTTPError(
-      `Row action '${rowActionId}' not found in '${tableId}'`,
-      400
-    )
+  return await updateDoc(tableId, rowActionId, actionsDoc => {
+    ensureUniqueAndThrow(actionsDoc, newName, rowActionId)
+    actionsDoc.actions[rowActionId].name = newName
+    return actionsDoc
+  })
+}
+
+async function guardView(tableId: string, viewId: string) {
+  let view
+  try {
+    view = await sdk.views.get(viewId)
+  } catch (e: any) {
+    if (e?.message !== "Unable to extract table ID, is not a view ID") {
```

Comment:
Not a fan on relying on error messages like this. Is there a function we could use to check if the ID is a valid view ID instead?

Reply:
I would agree - it likely makes sense to expose the `isViewID` function from the SDK so this can check the error and handle the scenario in its own unique way rather than relying on the exception.

Reply:
I agree that that's cleaner. Done!